### PR TITLE
Weather module

### PR DESCRIPTION
**Pull request by williamshoops96**
So I've adapted the original p #28  r and got an account on openweathermap to use the api key. I think, the intents need to be added on wit.ai but I'm not sure. Info on this? It's one of my first proper contributions to a project!

### CODE REVIEW

**Comment by swapagarwal** [comment]
@williamshoops96 Welcome to the world of open-source. Your PR looks quite good for a first contribution! :smile: 
The intents are already set up, that's why it passed the tests.
Just take a look at the lines I've commented on and this should be ready to merge!

**Comment by swapagarwal** [comment]
Remove all the .DS_Store files. There should be changes in 3 files only: https://github.com/swapagarwal/JARVIS-on-Messenger/pull/76/files

File: modules/src/__init__.py [hunk 1]
```diff
@@ -13,4 +13,5 @@
     'request',
     'time',
     'wiki',
+    'weather',
```
Comment (swapagarwal) [inline modules/src/__init__.py]: alphabetical

File: modules/src/weather.py [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+OPEN_WEATHER_MAP_ACCESS_TOKEN = os.environ.get('OPEN_WEATHER_MAP_ACCESS_TOKEN', config.OPEN_WEATHER_MAP_ACCESS_TOKEN)
+def process(input, entities):
+    output = {}
+
+    try:
+        location = entities['location'][0]['value']
+        r = requests.get('http://api.openweathermap.org/data/2.5/weather?q=' + location + '&appid=' + OPEN_WEATHER_MAP_ACCESS_TOKEN)
+        data = r.json()
+        output['input'] = input
+        template = TextTemplate('The weather in ' + data['name'] + 'is: ' + data['weather']['description'] + ' with a temperature of ' + data['main']['temp'] + '. Info provided by OpenWeatherMap.')
```
Comment (swapagarwal) [inline modules/src/weather.py]: Shouldn't it be `data['weather'][0]['description']`?

**Comment by williamshoops96** [comment]
Thank you haha! Program a lot, just trying to become better by
contributing! Okay I'll take a look later. I'm in school at the moment.

**Comment by williamshoops96** [comment]
And yeah sorry about the Ds store. Developing on mac, need to remember to
not commit those.

File: modules/src/__init__.py [hunk 3]
```diff
@@ -12,5 +12,7 @@
     'quote',
     'request',
     'time',
+    'weather',
     'wiki',
+
```
Comment (swapagarwal) [inline modules/src/__init__.py]: extra line

**Comment by swapagarwal** [comment]
Maybe this can be of help: https://jonbellah.com/recursively-remove-ds-store/
TL;DR: `find . -name '*.DS_Store' -type f -delete`

**Comment by williamshoops96** [comment]
Ah thanks. Will make it a lot faster!

**Comment by williamshoops96** [comment]
reference #79
